slasher/rpc/server.go

```diff
@@ -23,9 +23,28 @@ type Server struct {
 
 // IsSlashableAttestation returns an attester slashing if the attestation submitted
 // is a slashable vote.
-func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Attestation) (*ethpb.AttesterSlashing, error) {
-	//TODO(3133): implement attestation validation after attestation store will be merged.
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.IndexedAttestation) (*ethpb.AttesterSlashingResponse, error) {
+	//TODO(#3133): add signature validation
+	if err := ss.SlasherDB.SaveIndexedAttestation(req); err != nil {
+		return nil, err
+	}
+	tEpoch := req.Data.Target.Epoch
+	indices := append(req.CustodyBit_0Indices, req.CustodyBit_1Indices...)
+	root, err := ssz.HashTreeRoot(req.Data)
+	if err != nil {
+		return nil, err
+	}
+	atsSlashinngRes := &ethpb.AttesterSlashingResponse{}
+	for _, idx := range indices {
+		atts, err := ss.SlasherDB.DoubleVotes(tEpoch, idx, root[:])
+		if err != nil {
+			return nil, err
+		}
+		atsSlashinngRes.AttesterSlashing = append(atsSlashinngRes.AttesterSlashing, atts...)
```

Comment:
if `atts` is nil, you would be appending a nil object to the slice. You should validate it before appending it

Reply:
added, thanks